pkg/reconciler/events/storage/storage.go

```diff
@@ -82,22 +81,22 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, storage *v1alpha1.CloudS
 	ctx = logging.WithLogger(ctx, r.Logger.With(zap.Any("storage", storage)))
 
 	storage.Status.InitializeConditions()
-	storage.Status.ObservedGeneration = storage.Generation
 
 	topic := resources.GenerateTopicName(storage)
 	_, _, err := r.PubSubBase.ReconcilePubSub(ctx, storage, topic, resourceGroup)
 	if err != nil {
-		return reconciler.NewEvent(corev1.EventTypeWarning, "ReconcilePubSubFailed", "Failed to reconcile CloudStorageSource PubSub: %s", err)
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledPubSubFailed, "Failed to reconcile CloudStorageSource PubSub: %s", err)
 	}
 
 	notification, err := r.reconcileNotification(ctx, storage)
 	if err != nil {
-		storage.Status.MarkNotificationNotReady("NotificationReconcileFailed", "Failed to reconcile CloudStorageSource notification: %s", err.Error())
-		return reconciler.NewEvent(corev1.EventTypeWarning, "NotificationReconcileFailed", "Failed to reconcile CloudStorageSource notification: %s", err)
+		storage.Status.MarkNotificationNotReady(reconciledNotificationFailed, "Failed to reconcile CloudStorageSource notification: %s", err.Error())
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledNotificationFailed, "Failed to reconcile CloudStorageSource notification: %s", err)
 	}
 	storage.Status.MarkNotificationReady(notification)
+	storage.Status.ObservedGeneration = storage.Generation
```

Comment:
remove

Reply:
If I remove it from here and move it up top as you suggest I end up having to modify lots of failure test cases that do not pass as is. Specifically, I end up having to add lots of:
`WithCloudStorageSourceStatusObservedGeneration(generation),`
to these failure test cases. I moved this as suggested in https://github.com/google/knative-gcp/pull/605#discussion_r389199573. Lets talk with @ian-mi to see what the right place for this should be.

Reply:
done